### DISCUSSION TITLE
MultiQC

### DISCUSSION THREAD

**Discussion by gregdenay** [bio-raum/FooDMe2#82] [General]
Ich sehe kein Bedarf das MultiQC report zu behalten. Sollten wir es einfach komplett entfernen vom Workflow?

**Comment by marchoeppner** [comment]
Ja, kann man machen. Dann müssen wir nur schauen, welche Prozesse dann vielleicht obsolet sind.

**Comment by gregdenay** [comment]
I am somehow lost in nextflow again.

What I did:
`subworkflow /reporting/main.nf` :
```groovy
    emit:
    versions = ch_versions
    xlsx        = HELPER_REPORT_XLSX.out.xlsx
    report    = HELPER_HTML_REPORT.out.html
```

`workflows/foodme2nf``:
```groovy
    /*
    MULTIQC
    */
    // multiqc_files   = multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml)
    
    // MULTIQC(
    //     multiqc_files.collect(),
    //     ch_multiqc_config,
    //     ch_multiqc_logo
    // )

    emit:
    // qc= MULTIQC.out.html
    report = REPORTING.out.report
    xlsx = REPORTING.out.xlsx
```

What I get:
``` 
Missing process or function mix([DataflowStream[?], [DataflowGetPropertyExpression(value=null), DataflowGetPropertyExpression(value=null)]])

 -- Check script 'main.nf' at line: 39 or see '.nextflow.log' file for more details
```

I don't even know which `main.nf` I should check, line 39 in the reporting subworklow is empty, I don't understand what this error even means either.

Feels like I am missing something obvious?

**Comment by gregdenay** [comment]
Looks like I needed to create empty channels first...

Still running into an error at mysteirous line 39 though.
```
ERROR ~ No such variable: Exception evaluating property 'qc' for nextflow.script.ChannelOut, Reason: groovy.lang.MissingPropertyException: No such property: qc for class: groovyx.gpars.dataflow.DataflowBroadcast
```

I pushed it to the `multiqc_purge` branch if you want ot have a look.

**Reply by marchoeppner** [reply]
The "main" main.nf is the one in the base directory; specifically:

```NEXTFLOW
workflow {
    if (params.build_references) {
        BUILD_REFERENCES()
    } else {
        FOODME2()
        multiqc_report = multiqc_report.mix(FOODME2.out.qc).toList()
    }
}
```

I will admit that MultiQC is fairly "hard-wired" into the code template, all the way down to the Email reporting (which I am not using at the moment, since I cannot set up a mail service on my machine(s). 

Maybe too much effort to get rid off after all ? ;) Else you can just delete

```
multiqc_report = multiqc_report.mix(FOODME2.out.qc).toList()
```
to turn the multiqc reporting into an empty channel for the downstream email  stuff.

**Reply by marchoeppner** [reply]
The email stuff was actually a reason to keep MultiQC, since all the code is already in place and if you are so lucky to be able to run a mail service, it is super convenient to get the MultiQC report via Email automatically upon pipeline completion. But again, not the main report any more, so somewhat "optional":

**Reply by gregdenay** [reply]
🤡 it had to be something obvious.

I think I managed to switch mutliqc to the new report for the emailing, and cleanishly remove the multiqc module. There are still multiqc channels emitted in most subworkflows but it doesn't really matter. 
I don't have the luck of having a email server here, so I can't really test it.
